python/python/tests/test_optimize.py

```diff
@@ -36,6 +37,41 @@ def test_dataset_optimize(tmp_path: Path):
     assert dataset.version == 3
 
 
+def test_blob_compaction_requires_flag(tmp_path: Path):
+    base_dir = tmp_path / "blob_dataset"
+    blob_field = pa.field(
+        "blob", pa.large_binary(), metadata={"lance-encoding:blob": "true"}
+    )
+    schema = pa.schema([pa.field("id", pa.int32()), blob_field])
+    blobs = [b"\x01\x02", b"\x03\x04\x05"]
+    table = pa.table(
+        {
+            "id": pa.array([0, 1], type=pa.int32()),
+            "blob": pa.array(blobs, type=pa.large_binary()),
+        },
+        schema=schema,
+    )
+
+    dataset = lance.write_dataset(
+        table,
+        base_dir,
+        schema=schema,
+        max_rows_per_file=1,
+        data_storage_version="stable",
+    )
+    assert len(dataset.get_fragments()) == 2
```

Comment:
Does this measure anything meaningful? Aren't the blob fragments different than regular fragments?